source/CargoHold.cpp

```diff
@@ -562,6 +564,12 @@ int CargoHold::IllegalCargoFine() const
 	// Only the worst illegal outfit is fined.
 	for(const auto &it : outfits)
 	{
+		// The code for adding and removing outfits does not clear the entry in the
+		// map if its value becomes zero, so we need to check if the outfit is
+		// actually inside the cargo hold.
+		if(!it.second)
+			continue;
```

Comment:
Lets provide a custom container when calling `CargoHold::Outfits` to avoid needing to remember this. The container's iterator should skip outfits where the count is 0.

(A follow-up PR is fine.)

Reply:
This seems like more complexity than it's worth tbh. A whole container with custom iterators just for this seems overkill.